packages/plugins/build-report/src/xpack.ts

```diff
@@ -436,11 +470,14 @@ export const getXpackPlugin =
                         log.debug(`Could not find output of ${JSON.stringify(file)}`);
                         continue;
                     }
-
-                    if (outputFound.type !== 'map' && !entryOutputs.includes(outputFound)) {
-                        entryOutputs.push(outputFound);
+                    if (outputFound.type !== 'map' && !entryOutputs.has(outputFound.name)) {
+                        entryOutputs.set(outputFound.name, outputFound);
                         // We know it's not a map, so we cast it.
-                        entryInputs.push(...(outputFound.inputs as Input[]));
+                        for (const input of outputFound.inputs as Input[]) {
```

Comment:
why is the cast necessary here?

Reply:
In the case of sourcemaps, `file.inputs` can be `Outputs` as sourcemaps are built out of `Output` type of files.

So since I'm checking it's not a sourcemaps earlier, I'm casting it to make the rest of code type error free.

Very much looking for a better way.